### PR TITLE
Only test the prospective commit: ignore unstaged changes.

### PR DESCRIPTION
**Pull request by wimleers**
Prevents e.g. PHPUnit from tripping over `xdebug_break()` statements that you're not even trying to commit.

### CODE REVIEW

File: pre-commit [hunk 1]
```diff
@@ -13,6 +13,9 @@ else
     AGAINST=4b825dc642cb6eb9a060e54bf8d69288fbee4904
 fi
 
+# *Only* test the prospective commit: ignore unstaged changes.
+git stash save -q --keep-index
```
Comment (alexpott) [inline pre-commit]: I'm just wondering about problems caused by doing this on a branch with no changes:
 8.3.x  alex: ~/dev/drupal > git stash save -q --keep-index
 8.3.x  alex: ~/dev/drupal > git stash list
 8.3.x  alex: ~/dev/drupal > git stash pop -q
No stash found.

Imo we should only pop if something is stashed.

**Comment by alexpott** [comment]
I've always been unsure about this PR. Because it is super annoying that unstaged code can affect the checks - so that's wrong. But also I use these githooks on my regular code dev environment and often have unstaged code lying around.

**Comment by wimleers** [comment]
I completely lost track of this, because I stopped using `d8githooks` because the DX problem that this PR was trying to fix was getting in the way all the time for me. Picking this up again…

File: pre-commit (hunk 1 above)
[the remark this answers is not in the document]
Comment (wimleers) [reply pre-commit]: This was surprisingly hard to detect… fortunately https://stackoverflow.com/a/34116244 helped me :)

File: pre-commit [hunk 2]
```diff
@@ -77,6 +80,7 @@ for FILE in $FILES; do
 done
 
 if [ "$CHECK_PHPUNIT" == "1" ] ; then
+    ORIGINAL_DIR=`pwd`
```
Comment (wimleers) [inline pre-commit]: This seems out of scope?

**Comment by wimleers** [comment]
Rewritten solution :)

**Comment by alexpott** [comment]
I reverted this change it was causing issues with my work flow.

I kept on getting
```
Please commit your changes or stash them before you merge.
Aborting
The stash entry is kept in case you need it again.
```
Which is more annoying for me than the problem this is solving. We need to file another way
